src/plugins/testbed/server/index.ts

```diff
@@ -72,6 +72,26 @@ class Plugin {
       }
     );
 
+    router.get(
+      {
+        path: '/requestcontext/render/{id?}',
+        validate: {
+          params: schema.object({
+            id: schema.maybe(schema.string()),
+          }),
+        },
+      },
+      async (context, req, res) => {
+        const body = await context.core.rendering.render(req.params.id);
+        return res.ok({
+          body,
+          headers: {
+            'content-securty-policy': core.http.csp.directives,
+          },
```

Comment:
Question: I see that in `x-pack/plugins/security/server/routes/authentication/saml.ts` this header is also manually added. Shouldn't we add this by default on every response in core?

Reply:
CSP is only really useful for `text/html` responses, so maybe we could do it by default for all of those? Thoughts @restrry?

Reply:
Also, I have fixed the typo in this header locally.

Reply:
> CSP is only really useful for text/html responses, so maybe we could do it by default for all of those?

sounds good to me

Reply:
to note: suppresses https://github.com/elastic/kibana/pull/47375

Reply:
Sounds like we just need to allow endpoints to override it when needed, but can still have a good default policy.

Reply:
Suggesting to move this to #47375.